docs/development/visualize/development-create-visualization.asciidoc

```diff
@@ -0,0 +1,307 @@
+request handlers. Its also possible to define your own request handler 
+(which you can then register to be used by other visualizations)
+
+[[development-default-request-handler]]
+==== courier request handler
+'courier' is the default request handler which works with out 'default' side bar editor.
+
+[[development-none-request-handler]]
+==== `none` request handler
+Using 'none' as your request handles means your visualization does not require search.
+
+[[development-custom-request-handler]]
+==== custom request handler
+You can define your custom request handler by providing a function with the following definition:
+`function (vis, appState, uiState, searchSource) { ... }`
+
+[[development-visualization-response-handlers]]
+=== Visualization Response Handlers
+Response handler will receive the data from request handler and an instance of Vis object. Its job is to convert the data to a format visualization can use.
+by default 'none' request handler is used which just passes the data it receives back, but can be changed to 'basic' (which VislibVisTypeFactory uses) or to a custom one.
+
+[[development-default-response-handler]]
+==== default response handler
+Default response handler will convert pure elasticsearch response to tabular format.
+
+[[development-none-response-handler]]
+==== none response handler
+None response handler is an identity function, which will return the pure elasticsearch response.
+
+[[development-custom-response-handler]]
+==== custom response handler
+You can define your custom response handler by providing a function with the following definition:
+'function (vis, response) { ... }'
+
+[[development-vis-object]]
+=== Vis object
+Vis object holds the visualization state and is the window into kibana:
+
+- *vis.params*: holds the visualization parameters
+- *vis.indexPattern*: selected index pattern object
+- *vis.getState()*: gets current visualization state 
+- *vis.updateState()*: updates current state with values from `vis.params`
+- *vis.resetState()*: resets `vis.params` to the values in the current state
+- *vis.getUiState()*: gets UI state of visualization
+- *vis.uiStateVal(name, val)*: updates a property in UI state
+- *vis.isEditorMode()*: returns true if in editor mode
+- *vis.API.timeFilter*: allows you to access time picker
+- *vis.API.events.click*: default click handler
+- *vis.API.events.brush*: default brush handler
+
+Visualization gets all its parameters in `vis.params` (defaults merged with the current state). If it wants to update them it should update them in this object and then call `vis.updateState()` which will inform <visualize> about the change, which will call request and response handler and then your visualizations render method.
+
+For the parameters that should not be saved with visualization you should use UI state (such as current state: popup open/closed, custom colors applied to the series etc)
```

Comment:
Should custom colors applied to series, really not be saved with the visualization? It sounds like something, that would make sense to store in the visualization.